app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -0,0 +1,249 @@
+                                                   retrieveNino: NinoPredicate,
+                                                   val addIncomeSourceStartDate: AddIncomeSourceStartDate,
+                                                   val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                                   val retrieveBtaNavBar: NavBarPredicate,
+                                                   val customNotFoundErrorView: CustomNotFoundError,
+                                                   incomeSourceDetailsService: IncomeSourceDetailsService)
+                                                  (implicit val appConfig: FrontendAppConfig,
+                                                   implicit val itvcErrorHandler: ItvcErrorHandler,
+                                                   implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                   implicit val dateFormatter: ImplicitDateFormatterImpl,
+                                                   implicit val dateService: DateService,
+                                                   implicit override val mcc: MessagesControllerComponents,
+                                                   val ec: ExecutionContext)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching {
+
+  def showUKProperty: Action[AnyContent] = handleRequestMethod(UKProperty)(isGet = true)
+  def showUKPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(UKProperty)(isGet = true)
+  def showForeignProperty: Action[AnyContent] = handleRequestMethod(ForeignProperty)(isGet = true)
+  def showForeignPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(ForeignProperty)(isGet = true)
+  def showSoleTraderBusiness: Action[AnyContent] = handleRequestMethod(SoleTraderBusiness)(isGet = true)
+  def showSoleTraderBusinessAgent: Action[AnyContent] = handleRequestMethodAgent(SoleTraderBusiness)(isGet = true)
+
+  def submitUKProperty: Action[AnyContent] = handleRequestMethod(UKProperty)(isGet = false)
+  def submitUKPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(UKProperty)(isGet = false)
+  def submitForeignProperty: Action[AnyContent] = handleRequestMethod(ForeignProperty)(isGet = false)
+  def submitForeignPropertyAgent: Action[AnyContent] = handleRequestMethodAgent(ForeignProperty)(isGet = false)
+  def submitSoleTraderBusiness: Action[AnyContent] = handleRequestMethod(SoleTraderBusiness)(isGet = false)
+  def submitSoleTraderBusinessAgent: Action[AnyContent] = handleRequestMethodAgent(SoleTraderBusiness)(isGet = false)
+
+  private def handleRequestMethod(incomeSourceType: IncomeSourceType)
+                                 (isGet: Boolean): Action[AnyContent] =
+    (checkSessionTimeout andThen authenticate andThen retrieveNino
+      andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+      implicit user =>
+        if(isGet) {
```

Comment:
you can use request object to determine what type of request you have: GET or POST